### PR TITLE
Homepage products

### PR DESCRIPTION
**Pull request by OmarAlithawi**
PTAL

### CODE REVIEW

File: src/components/layout/productCard/ProductCard.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import heart from "../../../assets/svgs/heart.svg";
+import { Col } from "react-bootstrap";
+import "./style.scss";
+import heartBroken from "../../../assets/svgs/heart-broken.svg";
+
+export default function ProductCard(props) {
+  const [isLiked, setIsLiked] = React.useState(false);
+
+  const settings = {
+    className: "silderImages",
```
Comment (louisrli) [inline src/components/layout/productCard/ProductCard.jsx]: typo

File: src/components/layout/productCard/ProductCard.jsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import Slider from "react-slick";
+import heart from "../../../assets/svgs/heart.svg";
+import { Col } from "react-bootstrap";
+import "./style.scss";
+import heartBroken from "../../../assets/svgs/heart-broken.svg";
+
+export default function ProductCard(props) {
+  const [isLiked, setIsLiked] = React.useState(false);
```
Comment (louisrli) [inline src/components/layout/productCard/ProductCard.jsx:10]: is this supposed to be loaded from firebase or something eventually? or props?

File: src/components/layout/productCard/ProductCard.jsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          <img src={props.info.images[2]} alt="image" />
+        </div>
+      </Slider>
+    );
+  };
+
+  return (
+    <React.Fragment>
+      <Col xs={8} md={5} lg={3} className="productCard">
+        {/*doc-id = {props.productId[index]}*/}
```
Comment (louisrli) [inline src/components/layout/productCard/ProductCard.jsx]: commented code

File: src/components/layout/productCard/ProductCard.jsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        {likeBtnHandler()}
```
Comment (louisrli) [inline src/components/layout/productCard/ProductCard.jsx]: these are not handlers, usually handlers is something like `onClick={handleClick}`

for these, just call them

`const slider`
`const likeBtn` 

and they should not be functions

File: src/components/products/Products.jsx [hunk 5]
```diff
@@ -1,5 +1,80 @@
 import React from "react";
+import ProductCard from "../layout/productCard/ProductCard";
+import { Container, Row } from "react-bootstrap";
 
 export default function Products() {
-  return <div></div>;
+  const data = [
```
Comment (louisrli) [inline src/components/products/Products.jsx:8]: is this temporary until real products are placed? if so, add a comment indicating this

File: src/components/products/Products.jsx [hunk 6]
```diff
@@ -1,5 +1,80 @@
...
+
+  const renderProductsCards = () => {
+    return data.map((info, index) => {
+      return <ProductCard info={info} index={index} key={index} />;
+    });
+  };
+
+  return (
+    <Container>
+      <Row>{renderProductsCards()}</Row>
```
Comment (louisrli) [inline src/components/products/Products.jsx]: just inline this map code, it's short enough that having the function is probably a bit more confusing

**Comment by OmarAlithawi** [comment]
PTAL

**Review by MiraAr — APPROVED** [review]

File: src/components/layout/productCard/ProductCard.jsx [hunk 7]
```diff
@@ -6,10 +6,11 @@ import "./style.scss";
 import heartBroken from "../../../assets/svgs/heart-broken.svg";
 
 export default function ProductCard(props) {
+  // we will load isLiked from firebase and then set it to this state
```
Comment (louisrli) [inline src/components/layout/productCard/ProductCard.jsx]: capitalize we

File: src/components/layout/productCard/ProductCard.jsx [hunk 8]
```diff
@@ -20,51 +21,35 @@ export default function ProductCard(props) {
...
-  };
-
   return (
     <React.Fragment>
-      <Col xs={8} md={5} lg={3} className="productCard">
-        {/*doc-id = {props.productId[index]}*/}
-        {likeBtnHandler()}
-        {sliderHandler()}
-
+      <Col xs={8} md={5} lg={3} className="productCard"> {/*when we get the real product we need to add an id to each product*/}
```
Comment (louisrli) [inline src/components/layout/productCard/ProductCard.jsx]: don't do comments at end of line like this, it's ugly, put it after line 25

File: src/components/layout/productCard/ProductCard.jsx [hunk 9]
```diff
@@ -20,51 +21,35 @@ export default function ProductCard(props) {
...
+        {isLiked ? (
```
Comment (louisrli) [inline src/components/layout/productCard/ProductCard.jsx]: so for these you should actually include them outside as variables, it makes the code cleaner. but i was saying that they should not be functions, they are simply variables.

```const slider = {
  return()
}
```
(in your old code you had `slider = () =>`

**Review by louisrli — CHANGES_REQUESTED** [review]

**Comment by OmarAlithawi** [comment]
PTAL

File: src/components/products/Products.jsx [hunk 10]
```diff
@@ -1,5 +1,80 @@
 import React from "react";
+import ProductCard from "../layout/productCard/ProductCard";
+import { Container, Row } from "react-bootstrap";
 
 export default function Products() {
-  return <div></div>;
+  // This is a temporary until we add real products in the database
+
```
Comment (louisrli) [inline src/components/products/Products.jsx:7]: line 7 delete newline

**Review by louisrli — APPROVED** [review]

**Comment by louisrli** [comment]
fix merge conflicts and i believe you can merge yourself with approving reviews

**Pull request merged**